lib/PhasarLLVM/ControlFlow/LLVMBasedICFG.cpp

```
@@ -423,6 +425,61 @@ bool LLVMBasedICFG::constructDynamicCall(const llvm::Instruction *I,
   return NewTargetsFound;
 }
 
+bool LLVMBasedICFG::addRuntimeEdges(
+    const std::set<std::pair<unsigned, unsigned>> &CallerCalleeMap) {
+  bool ICFGAugumented = false;
+  for (const auto &[CallSiteId, FunctionPrefixId] : CallerCalleeMap) {
+    const auto *CallSite = IRDB.getInstruction(CallSiteId);
+    assert(CallSite && "addRuntimeEdges: callSite not found in IRDB");
+    const auto *CallBase = llvm::dyn_cast<llvm::CallBase>(CallSite);
+    if (!CallBase) {
+      continue;
+    }
+    const auto *RuntimeCallTarget = IRDB.getFunctionById(FunctionPrefixId);
+    if (!RuntimeCallTarget) {
+      /// TODO: At least log this incident
+      continue;
+    }
+    vertex_t CallSiteVertexDescriptor;
+    auto FvmItr = FunctionVertexMap.find(CallSite->getFunction());
+    if (FvmItr != FunctionVertexMap.end()) {
+      CallSiteVertexDescriptor = FvmItr->second;
+    } else {
+      LOG_IF_ENABLE(BOOST_LOG_SEV(lg::get(), ERROR)
+                    << "addRuntimeEdges: Did not find vertex of "
+                       "calling function "
+                    << CallSite->getFunction()->getName().str()
+                    << " at callsite " << llvmIRToString(CallSite));
+      std::terminate();
```

Why do we need to fail hard here? Couldn't we just skip this entry?